src/FSharp.Core/map.fs

```diff
@@ -64,373 +72,483 @@ module MapTree =
     let mutable largestMapSize = 0
     let mutable largestMapStackTrace = Unchecked.defaultof<_>
 
-    let report() = 
-       traceCount <- traceCount + 1 
-       if traceCount % 1000000 = 0 then 
-           System.Console.WriteLine(
-               "#MapOne = {0}, #MapNode = {1}, #Add = {2}, #Remove = {3}, #Unions = {4}, #Lookups = {5}, avMapTreeSizeOnNodeCreation = {6}, avMapSizeOnCreation = {7}, avMapSizeOnLookup = {8}", 
-               numOnes, numNodes, numAdds, numRemoves, numUnions, numLookups, 
-               (totalSizeOnNodeCreation / float (numNodes + numOnes)), (totalSizeOnMapAdd / float numAdds), 
-               (totalSizeOnMapLookup / float numLookups))
-           System.Console.WriteLine("#largestMapSize = {0}, largestMapStackTrace = {1}", largestMapSize, largestMapStackTrace)
-
-    let MapTree (k,v) = 
-        report()
+    let report () =
+        traceCount <- traceCount + 1
+
+        if traceCount % 1000000 = 0 then
+            System.Console.WriteLine(
+                "#MapOne = {0}, #MapNode = {1}, #Add = {2}, #Remove = {3}, #Unions = {4}, #Lookups = {5}, avMapTreeSizeOnNodeCreation = {6}, avMapSizeOnCreation = {7}, avMapSizeOnLookup = {8}",
+                numOnes,
+                numNodes,
+                numAdds,
+                numRemoves,
+                numUnions,
+                numLookups,
+                (totalSizeOnNodeCreation
+                 / float (numNodes + numOnes)),
```

Comment:
Is it expected to have line break here, in the middle of sub-expression? It makes it a tad more difficult to read.

Reply:
From Fantomas point of view, this is an infix expression that is longer than `fsharp_max_infix_operator_expression`.
Fantomas does not know what your code does, so applies the rules it has.
I don't think Fantomas respects https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-arithmetic-and-binary-expressions.

Reply:
The wider line limit fixed this issue for this particular code.